interview-questions.md

```diff
@@ -7,44 +7,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: The foreign key always goes on the belongs_to side. To create the association between the tweo models by adding in the Cohort model file add "has_many" and add "belongs_to" in the Student model file.
 
 
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: edit, destroy, and show, becasue those are the ones that modify things. The params usually has .find attached to it because we are trying to display one piece of particular data.
 
-  Researched answer:
+  Researched answer: Show, update, destroy, edit. These are the ones that need params to be passed because we are calling on a specific piece of data and we need to know what we are calling on.
 
 
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: rails genereate model which creates the model, rails generate resource which creates the model and the controller, and rails generate controller which generates the controller.
 
-  Researched answer:
+  Researched answer: Rails generate model makes the model by passing the model name and here is wher you can also include arguments or names of the columns in your database. Rails generate resource creates the model and the controller and automatically create the routes, migration file, controller file, and pipeline files. Rails generate controller will create the controller and also the ciontroller file, test file, helper for the view, view sheet, and stylesheet file.
 
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET"    location: /students          
+action: "GET"    location: /students    
+index: which would show all the students in the database      
 
-action: "POST"   location: /students       
+action: "POST"   location: /students     
+create: which adds new info into the database  
 
 action: "GET"    location: /students/new
+new: this displays a form to the user to put new entries into the database
 
 action: "GET"    location: /students/2  
+show: which will display to the uder student with id 2
 
 action: "GET"    location: /students/2/edit    
+edit: which will show a form to the user to edit data in an existing student
 
 action: "PATCH"  location: /students/2      
+update: this will modify the data for student with id 2
 
-action: "DELETE" location: /students/2      
+action: "DELETE" location: /students/2    
+destroy: this would delete the student with id 2  
 
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```

Comment:
Awesome!  I see all the CRUD actions here which is the foundation of a solid working app!